crates/cow-amm/src/maintainers.rs

```diff
@@ -0,0 +1,36 @@
+use {
+    crate::cache::Storage,
+    ethcontract::futures::future::join_all,
+    ethrpc::Web3,
+    shared::maintenance::Maintaining,
+    std::sync::Arc,
+    tokio::sync::RwLock,
+};
+
+pub struct AmmTokenBalanceMaintainer {
```

Comment:
naming: we are already in the context of cow amm so we can drop that. What does TokenBalanceMaintaining mean (we are not maintaining a balance)? 

Isn't this more something like Pruning or if we want to be more explicit EmptyPoolRemoval or the like?